xdsl/dialects/pdl_interp.py

```diff
@@ -273,6 +276,30 @@ def __init__(
             result_types=[result_type],
         )
 
+    @classmethod
+    def parse(cls, parser: Parser) -> GetResultsOp:
+        index = parser.parse_optional_integer()
+        if index is not None:
+            index = IntegerAttr.from_int_and_width(index, 32)
+        parser.parse_characters("of")
+        input_op = parser.parse_operand()
+        parser.parse_punctuation(":")
+        result_type = parser.parse_type()
+        return GetResultsOp.build(
+            operands=(input_op,),
+            properties={"index": index},
+            result_types=(result_type,),
+        )
+
+    def print(self, printer: Printer):
+        if self.index is not None:
+            printer.print(" ")
+            printer.print(self.index.value.data)
+        printer.print(" of ")
+        printer.print_operand(self.input_op)
+        printer.print(" : ")
+        printer.print(self.value.type)
```

Comment:
More of a nit, but for the future I'd recommend specific APIs that aren't doing dispatch based on type:
```suggestion
        if self.index is not None:
            printer.print_string(" ", indent=0)
            printer.print_operand(self.index.value.data)
        printer.print_string(" of ", indent=0)
        printer.print_operand(self.input_op)
        printer.print_string(" : ", indent=0)
        printer.print_attribute(self.value.type)
```
The indent=0 removes the check for newlines in the input string and prints it as-is, instead of calculating indentation levels.